llvm/plan.sh

```diff
@@ -32,44 +32,78 @@ do_unpack() {
   # There may be some more awesome way to do this - I don't know that yet.
   build_line "Unpacking $pkg_filename to custom cache dir"
   local source_file=$HAB_CACHE_SRC_PATH/$pkg_filename
-  local unpack_dir=$HAB_CACHE_SRC_PATH/${pkg_name}-${pkg_version}
+  # we want to keep the src as some later dependencies require it
+  local unpack_dir="${pkg_prefix}/src"
   mkdir -p "$unpack_dir"
-  pushd "$unpack_dir" > /dev/null
+  pushd "$unpack_dir" > /dev/null || exit
   # Per tar's help output:
   #
   #   --no-same-owner        extract files as yourself (default for ordinary users)
   #
   # The llvm package has some files owned by specific UIDs that we
   # can't be sure exist on the builder or target system.
   tar xf "$source_file" --strip 1 --no-same-owner
-  popd > /dev/null
-
-  # Download Clang frontend and place it in the correct place
-  build_line "Unpacking Clang FrontEnd to custom cache dir"
-  download_file http://llvm.org/releases/${pkg_version}/cfe-${pkg_version}.src.tar.xz \
-    cfe-${pkg_version}.src.tar.xz \
-    e6c4cebb96dee827fa0470af313dff265af391cb6da8d429842ef208c8f25e63
-
-  local clang_src_dir="$unpack_dir/tools/clang"
-  mkdir -p "$clang_src_dir"
-  pushd "$clang_src_dir" > /dev/null
-  tar xf "$HAB_CACHE_SRC_PATH/cfe-${pkg_version}.src.tar.xz" --strip 1 --no-same-owner
-  popd > /dev/null
+  popd > /dev/null || exit
 }
 
 do_build() {
   mkdir -p build
   cd build || exit
-  cmake -DCMAKE_INSTALL_PREFIX:PATH="$pkg_prefix" -G "Ninja" ../
-  ninja
+  cmake \
+    -DCMAKE_INSTALL_PREFIX:PATH="$pkg_prefix" \
+    -DCMAKE_BUILD_TYPE=Release \
+    -G "Unix Makefiles" "${pkg_prefix}/src"
+  make
 }
 
 do_check() {
   cd build || exit
-  ninja check
+  make test
 }
 
 do_install() {
   cd build || exit
-  ninja install
+  make install
+
+  # fix the interpreters in the `src`
+  _fix_interpreter_in_path "$pkg_prefix/src" '*.py' core/python2 bin/python
+  _fix_interpreter_in_path "$pkg_prefix/src" '*.py' core/coreutils bin/env
+  _fix_interpreter_in_path "$pkg_prefix/src" '*.sh' core/coreutils bin/env
+}
+
+do_strip() {
+  build_line "Stripping unneeded symbols from binaries and libraries"
+  # we need to skip the src folder
+  for folder in bin include lib; do
+    find "$pkg_prefix/$folder" -type f -perm -u+w -print0 2> /dev/null \
+      | while read -rd '' f; do
+        case "$(file -bi "$f")" in
+          *application/x-executable*) strip --strip-all "$f";;
+          *application/x-sharedlib*) strip --strip-unneeded "$f";;
+          *application/x-archive*) strip --strip-debug "$f";;
+          *) continue;;
+          esac
+      done
+  done
+}
+
+# private #
```

Comment:
![](https://media2.giphy.com/media/Kc7DXpUjFWAzm/200w.gif)

Reply:
😜 not quite private ... but, the `_` prefix in functions has been something I have always done.